content/tutorials/dsx-subscribe.textile

```diff
@@ -0,0 +1,249 @@
+    let newBodyTfl = document.createElement('tbody');
+    arrivals.forEach((arrival) => {
+      let arrivalTime = new Date(arrival.ExpectedArrival).toLocaleTimeString();
+      let tmpTableRow = newBodyTfl.insertRow(0);
+      tmpTableRow.insertCell(0).innerHTML = arrivalTime;
+      tmpTableRow.insertCell(1).innerHTML = arrival.Towards;
+      tmpTableRow.insertCell(2).innerHTML = arrival.PlatformName;
+      tmpTableRow.insertCell(3).innerHTML = arrival.LineName;
+    });
+    tableBodyTfl.parentNode.replaceChild(newBodyTfl, tableBodyTfl);
+    tableBodyTfl = newBodyTfl;
+  }
+```
+
+As you can see, the @updateTfl()@ function updates the table whenever a new update is received on the channel. We are all set. However, do you see the data changing? It's surely hard to observe the change. Let's add a minor style detail that flashes a color over the table when any data is updated, giving us a better visual way to see realtime data in action. 
+
+Go ahead and add the following code in a new CSS file and name it @flashbox.css@
+
+```[css]
+.c-flashbox {
+    display: inline-block;
+    overflow: auto;
+}
+
+.c-flashbox__positive-flash {
+    -webkit-animation-name: positive-flash;
+    -webkit-animation-duration: 3000ms;
+    -webkit-animation-iteration-count: 1;
+    -webkit-animation-timing-function: ease-in-out;
+}
+
+@-webkit-keyframes positive-flash {
+    0% {
+        background-color: White;
+    }
+    22% {
+        background-color: Green;
+    }
+    77% {
+        background-color: Green;
+    }
+    100% {
+        background-color: White;
+    }
+}
+```
+
+The two final steps would be to reference this CSS from our HTML file and add a bit of logic in the JavaScript to highlight new bits of data with a flash.
+
+```[html]
+<link rel="stylesheet" type="text/css" href="flashbox.css">
+```
+
+```[javascript]
+  function updateTfl(arrivals) {
+    arrivals = arrivals.reverse();
+    let newBodyTfl = document.createElement('tbody');
+    arrivals.forEach((arrival) => {
+      let arrivalTime = new Date(arrival.ExpectedArrival).toLocaleTimeString();
+      let tmpTableRow = newBodyTfl.insertRow(0);
+      tmpTableRow.insertCell(0).innerHTML = arrivalTime;
+      tmpTableRow.insertCell(1).innerHTML = arrival.Towards;
+      tmpTableRow.insertCell(2).innerHTML = arrival.PlatformName;
+      tmpTableRow.insertCell(3).innerHTML = arrival.LineName;
+    });
+
+    //new flashbox added here
+    for(let i=0; i < arrivals.length; i++) {
+        let arrivalTime = new Date(arrivals[arrivals.length - i - 1].ExpectedArrival).toLocaleTimeString();
+        if(tableBodyTfl.rows[i] != undefined && newBodyTfl.rows[i] != undefined){
+          if(tableBodyTfl.rows[i].cells[0].innerHTML != arrivalTime) {
+            newBodyTfl.rows[i].classList.add("c-flashbox__positive-flash");
+          }
+        }
+      }
+
+    tableBodyTfl.parentNode.replaceChild(newBodyTfl, tableBodyTfl);
+    tableBodyTfl = newBodyTfl;
+  }
+```
+
+h2. Next steps
+
```

Comment:
Could be good to have a bit saying 'see this live on our examples page!' linking to it.